src/RunTests/RunVSTestTask.cs

```diff
@@ -0,0 +1,443 @@
+                }
+
+                allArgs.Add(blameArgs);
+            }
+
+            if (VSTestCollect != null && VSTestCollect.Length > 0)
+            {
+                foreach (var arg in VSTestCollect)
+                {
+                    // For collecting code coverage, argument value can be either "Code Coverage" or "Code Coverage;a=b;c=d".
+                    // Split the argument with ';' and compare first token value.
+                    var tokens = arg.Split(';');
+
+                    if (arg.Equals(CodeCoverageString, StringComparison.OrdinalIgnoreCase) ||
+                        tokens[0].Equals(CodeCoverageString, StringComparison.OrdinalIgnoreCase))
+                    {
+                        isCollectCodeCoverageEnabled = true;
+                    }
+
+                    allArgs.Add("--collect:" + ArgumentEscaper.HandleEscapeSequenceInArgForProcessStart(arg));
+                }
+            }
+
+            if (isCollectCodeCoverageEnabled || isRunSettingsEnabled)
+            {
+                // Pass TraceDataCollector path to vstest.console as TestAdapterPath if --collect "Code Coverage"
+                // or --settings (User can enable code coverage from runsettings) option given.
+                // Not parsing the runsettings for two reason:
+                //    1. To keep no knowledge of runsettings structure in VSTestTask.
+                //    2. Impact of adding adapter path always is minimal. (worst case: loads additional data collector assembly in datacollector process.)
+                // This is required due to currently trace datacollector not ships with dotnet sdk, can be remove once we have
+                // go code coverage x-plat.
+                if (!string.IsNullOrEmpty(VSTestTraceDataCollectorDirectoryPath))
+                {
+                    allArgs.Add("--testAdapterPath:" +
+                                ArgumentEscaper.HandleEscapeSequenceInArgForProcessStart(
+                                    VSTestTraceDataCollectorDirectoryPath));
+                }
+            }
+
+            if (!string.IsNullOrEmpty(VSTestNoLogo))
+            {
+                allArgs.Add("--nologo");
+            }
+
+            if (!string.IsNullOrEmpty(VSTestArtifactsProcessingMode) && VSTestArtifactsProcessingMode.Equals("collect", StringComparison.OrdinalIgnoreCase))
+            {
+                allArgs.Add("--artifactsProcessingMode-collect");
+            }
+
+            if (!string.IsNullOrEmpty(VSTestSessionCorrelationId))
+            {
+                allArgs.Add("--testSessionCorrelationId:" + ArgumentEscaper.HandleEscapeSequenceInArgForProcessStart(VSTestSessionCorrelationId));
+            }
+
+            return allArgs;
+        }
+
+        private Task<int> ExecuteTest()
+        {
+#if NET6_0_OR_GREATER
+            string packagePath = $@"{NugetPath}\packages\microsoft.testplatform\{VSTestRunnerVersion}\tools\net6.0\Common7\IDE\Extensions\TestPlatform\";
```

Comment:
Seems like this strongly depends on the TF of Microsoft.TestPlatform. What if it changes? Will people need to coordinate this package version with the Microsoft.TestPlatform package one? If so, that should be documented and a table of compatible versions should be published (ie "if you're using Microsoft.TestPlatform version X, use this version of this package")